conversion/conversion.go

```diff
@@ -188,7 +194,12 @@ func dataFromDatabase(sourceProfile profiles.SourceProfile, config writer.BatchW
 	p := internal.NewProgress(totalRows, "Writing data to Spanner", internal.Verbose(), false)
 	rows := int64(0)
 	config.Write = func(m []*sp.Mutation) error {
-		_, err := client.Apply(context.Background(), m)
+		migrationData := migration.MigrationData{
+			MigrationRequestId: conv.MigrationData.MigrationRequestId,
+		}
+		serializedMigrationData, _ := proto.Marshal(&migrationData)
+		migrationMetadataValue := base64.StdEncoding.EncodeToString(serializedMigrationData)
+		_, err := client.Apply(metadata.AppendToOutgoingContext(context.Background(), migrationMetadataKey, migrationMetadataValue), m)
```

Comment:
This code is duplicated multiple times in this file and it seems like a lot of adjacent code is also duplicated. Can you factor this out in a separate method preferably the entire adjacent duplicate code but if its too much then at least the new code that we are adding.